instrumentation/net/http/otelhttp/test/handler_test.go

```diff
@@ -80,6 +82,17 @@ func assertScopeMetrics(t *testing.T, sm metricdata.ScopeMetrics, attrs attribut
 	assert.Equal(t, attrs, dPt.Attributes, "attributes")
 	assert.Equal(t, uint64(1), dPt.Count, "count")
 	assert.Equal(t, []float64{0, 5, 10, 25, 50, 75, 100, 250, 500, 750, 1000, 2500, 5000, 7500, 10000}, dPt.Bounds, "bounds")
+
+	want = metricdata.Metrics{
+		Name:        "http.server.request_count",
```

Comment:
Is there anything in the OTel specification about this metric? I cannot find it in https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/http.md

Shoudn't we deprecate/remove `RequestCount` and possibly other non-spec-compliant metrics as well?